### PR TITLE
docs(auth-kit): Update docs for the new `SafeAuthPack`

### PR DESCRIPTION
**Pull request by dasanra**
## What it solves
This PR updates the docs for the `auth-kit`:

- https://github.com/safe-global/safe-docs/issues/262
- https://github.com/safe-global/safe-docs/issues/263

## Other info
 - Continuation of https://github.com/safe-global/safe-docs/pull/273

### CODE REVIEW

File: safe-core-sdk/auth-kit/guides/safe-auth.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
+# Integration with SafeAuth
+
+This guide demonstrates creating an externally-owned account using your email or social media account. Once authenticated, you can sign transactions and interact with your Safe accounts.
+
+The `SafeAuthPack` is an authentication system that utilizes the [Web3Auth](https://web3auth.io) MPC technology. It was developed in collaboration with Web3Auth to create a smooth onboarding experience for web2 users across different dApps.
```
Comment (tanay1337) [inline safe-core-sdk/auth-kit/guides/safe-auth.md:5]: Could you add onboarding to Vale's allowlist?

Comment (tanay1337) [inline safe-core-sdk/auth-kit/guides/safe-auth.md:5]:
```suggestion
The `SafeAuthPack` is an authentication system that utilizes the [Web3Auth](https://web3auth.io) MPC technology. It was developed in collaboration with Web3Auth to create a smooth onboarding experience for web2 users across different dapps.
```

File: safe-core-sdk/auth-kit/guides/safe-auth.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+### Install dependencies
+
+```bash
+yarn add @safe-global/auth-kit@alpha @web3auth/safeauth-embed
+```
+
+## Create a SafeAuthPack instance
+
+We will use the provided `SafeAuthPack` exported in the `@safe-global/auth-kit` package.
```
Comment (tanay1337) [inline safe-core-sdk/auth-kit/guides/safe-auth.md:19]:
```suggestion
We will use the `SafeAuthPack` exported from the `@safe-global/auth-kit` package.
```

File: safe-core-sdk/auth-kit/guides/safe-auth.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+Create an instance of the [SafeAuthPack](https://github.com/safe-global/safe-core-sdk/tree/main/packages/auth-kit/src/packs/safe-auth/SafeAuthPack.ts) using the required `SafeAuthConfig` configuration object.
+
+> The list of supported chains is:
```
Comment (tanay1337) [inline safe-core-sdk/auth-kit/guides/safe-auth.md:23]:
```suggestion
> Supported networks:
```

File: safe-core-sdk/auth-kit/guides/safe-auth.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+await safeAuthPack.signOut()
+```
+
+After the user is authenticated, call `getProvider()` to get the Ethereum provider instance. This is a [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) compatible provider you can wrap using your favorite library (web3, ethers).
+
+```typescript
+safeAuthPack.getProvider()
+```
+
+We expose two methods for listening to events, `subscribe()` and `unsubscribe()`.
```
Comment (tanay1337) [inline safe-core-sdk/auth-kit/guides/safe-auth.md:89]:
```suggestion
We offer two methods for listening to events, `subscribe()` and `unsubscribe()`.
```

File: safe-core-sdk/auth-kit/reference/AuthKitBasePack.md [hunk 5]
```diff
@@ -20,23 +20,27 @@ class MyPack extends AuthKitBasePack {
 }
 ```
 
-## Abstract methods that Auth packs have to implement
+## Abstract methods and properties that any pack must implement
 
-These methods are the common interface for all the Auth packs. Check each pack's documentation to get more details.
+These methods and properties are the common interface for all the Auth packs. Check each pack's documentation to get more details.
```
Comment (tanay1337) [inline safe-core-sdk/auth-kit/reference/AuthKitBasePack.md:25]:
```suggestion
These methods and properties are the common interface for all the Auth packs. Check each pack's documentation for more details.
```

File: safe-core-sdk/auth-kit/reference/AuthKitBasePack.md [hunk 6]
```diff
@@ -20,23 +20,27 @@ class MyPack extends AuthKitBasePack {
...
 
-### `init(options?)`
+### `isAuthenticated(): boolean`
+
+Any pack extending the `AuthKitBasePack` class must implement the `isAuthenticated()` getter. This method returns a `boolean` indicating if the user is authenticated or not.
```
Comment (tanay1337) [inline safe-core-sdk/auth-kit/reference/AuthKitBasePack.md:29]:
```suggestion
Any pack extending the `AuthKitBasePack` class must implement the `isAuthenticated()` getter. This method returns a `boolean` indicating whether the user is authenticated.
```

File: safe-core-sdk/auth-kit/reference/SafeAuthPack.md [hunk 7]
```diff
@@ -0,0 +1,220 @@
+# SafeAuthPack
+
+SafeAuth is a pluggable authentication infrastructure for Web3 wallets and applications. It simplifies onboarding for mainstream and crypto native users, offering experiences tailored to their preferences. It supports all social logins, web and mobile native platforms, wallets, and other key management methods, creating a cryptographic key provider specific to the user and application.
```
Comment (tanay1337) [inline safe-core-sdk/auth-kit/reference/SafeAuthPack.md:3]: Could you add pluggable to the Vale allow list?

Comment (tanay1337) [inline safe-core-sdk/auth-kit/reference/SafeAuthPack.md:3]:
```suggestion
SafeAuth is a pluggable authentication infrastructure for web3 wallets and applications. It simplifies onboarding for mainstream and crypto native users, offering experiences tailored to their preferences. It supports all social logins, web and mobile native platforms, wallets, and other key management methods, creating a cryptographic key provider specific to the user and application.
```

File: safe-core-sdk/auth-kit/reference/SafeAuthPack.md [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+
+The `SafeAuthPack` enables using the [Web3Auth infrastructure](https://web3auth.io) with added Safe capabilities, such as retrieving the related Safe addresses for a given externally-owned account created using SafeAuth services.
+
+The pack helps onboard web2 users and provides an Ethereum signer address that can be used across any dApp using the different Ethereum chains.
```
Comment (tanay1337) [inline safe-core-sdk/auth-kit/reference/SafeAuthPack.md:7]:
```suggestion
The pack helps onboard web2 users and provides an Ethereum signer address that can be used across any dapp using the different Ethereum chains.
```

File: safe-core-sdk/auth-kit/reference/SafeAuthPack.md [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+```typescript
+const safeAuthPack = new SafeAuthPack({
+  txServiceUrl: 'https://safe-transaction-mainnet.safe.global',
+});
+await safeAuthPack.init(safeAuthInitOptions);
+```
+
+### `new SafeAuthPack(safeAuthConfig)`
+
+**Params**
```
Comment (tanay1337) [inline safe-core-sdk/auth-kit/reference/SafeAuthPack.md:30]:
```suggestion
**Parameters**
```

File: safe-core-sdk/auth-kit/reference/SafeAuthPack.md [hunk 10]
```diff
@@ -0,0 +1,220 @@
...
+
+- `safeAuthConfig?` - The `SafeAuthPack` class instantiation accepts the following options in its configuration:
+
+```typescript
+SafeAuthConfig {
+  txServiceUrl?: string
+}
+```
+
+- `txServiceUrl?` - The URL of the Safe Transaction Service. It's mandatory in chains where Safe doesn't provide a Transaction Service. It's used to retrieve the Safe addresses for an EOA created using SafeAuth services.
```
Comment (tanay1337) [inline safe-core-sdk/auth-kit/reference/SafeAuthPack.md:40]:
```suggestion
- `txServiceUrl?` - The URL of the Safe Transaction Service. It's mandatory in chains where Safe doesn't provide a Transaction Service. It retrieves the Safe addresses for an EOA created using SafeAuth services.
```

**Review by yagopv — APPROVED** [review]

**Review by tanay1337 — CHANGES_REQUESTED** [review]

File: safe-core-sdk/auth-kit/guides/safe-auth.md (hunk 1 above)
Comment (yagopv) [reply safe-core-sdk/auth-kit/guides/safe-auth.md]: done

**Pull request merged**